tests/test_project.py

```diff
@@ -1982,5 +1983,17 @@ def test_pickle_jobs_directly(self):
             self.assertEqual(pickle.loads(pickle.dumps(job)), job)
 
 
+class TestTestingProjectInitialization(unittest.TestCase):
+
+    # Sanity check on all different combinations of inputs
+    def test_input_args(self):
+        for nested, listed, het in itertools.product([True, False], repeat=3):
+            with TemporaryDirectory(prefix='signac_') as _tmp_dir:
+                root = _tmp_dir
+                pr = signac.init_project(name='testproject', root=root)
+                pr = signac.testing.init_jobs(pr, nested=nested, listed=listed,
+                        heterogeneous=het)
```

Comment:
Can we add the following sanity tests here?
1. Test the length of the project before (0) and after(N).
2. Test identity of the return value.
3. Call `pr.detect_schema()`.

Reply:
I added in those tests. The last fails because of issue #217

Reply:
Thx!